### PR TITLE
Stop user values reaching the filesystem and the generated code raw

### PR DESCRIPTION
**Pull request by NaokiTsuchiya**
## Problem

A dependency index is `Interface-qualifier`. Only `\` was replaced when building the script file name, so a qualifier from `->annotatedWith()` reached the filesystem raw — and both the index and `->toInstance()` values reached the generated code raw.

- `annotatedWith('a/b')` writes a directory, not a flat script. Once `Interface-x/` exists, a later `x/../../escaped` makes `rename()` write outside the script dir — and the read side rebuilds the same unnormalised path, so it still resolves. Anything treating the script dir as the whole artifact misses it.
- A NUL byte aborts the compile with a raw `ValueError` from `rename()`.
- `'` closes the literal holding the index, and what follows is compiled as PHP in a scope holding `$scriptDir` and `$singletons`. A trailing `\` escapes the closing quote instead, and the script stops parsing. (`$` alone is inert — the literal is single-quoted.)
- `toInstance(['k' => "it's"])` does the same to the literal holding the `serialize()` output: the compiled script is no longer PHP.

## Fix

`ScriptName::from()` percent-encodes every byte outside `[A-Za-z0-9_.-]` after the existing `\`→`_`. The name stays one path segment, and an encoded index can never land on an unencoded one's file. Write side, read side and generated code share it.

`var_export()` replaces both raw interpolations. No user-controlled value is interpolated into generated code any more; what is left is class and method names, which cannot carry a quote.

## Compatibility

Fixtures here compile to byte-identical file names (26 scripts). Only an index holding a byte outside `[A-Za-z0-9_.\-\x80-\xFF]` is renamed — exactly the broken set. Generated scripts change only by `\`→`\\` inside the index literal, and inside a serialized value that holds a quote or a backslash; the same strings either way.

Slash qualifiers work today, just not flat, so this needs a recompile, and so does a rollback. Minor, not patch.

## Not fixed

`\`→`_` predates this and is not injective: `annotatedWith(Foo\Bar::class)` shares a file with `annotatedWith('Foo_Bar')`. Encoding `\` renames every script, encoding `_` every snake_case qualifier, and the collision takes two bindings spelling one name two ways. Characterised in `ScriptNameTest`.

## Tests

Each of the five ways Ray.Di delivers a qualifier, plus instance values. All three commits green on their own.

## Summary by CodeRabbit

* **New Features**
  * Added validated, filesystem-safe script naming.
  * Improved qualifier-path handling, including trailing and consecutive backslashes.
* **Bug Fixes**
  * Invalid qualifiers now provide clearer exception context and surface as unbound dependencies at runtime.
  * Hardened generated code against escaping issues in instance values.
* **Tests**
  * Added coverage for script naming, qualifier edge cases, and instance-value escaping.

### CODE REVIEW

File: src/InstanceScript.php [hunk 1]
```diff
@@ -97,18 +96,19 @@ private function addDependencyArg(bool $isSingleton, string $index, ReflectionPa
...
         $ip = sprintf("['%s', '%s', '%s']", $parameter->getDeclaringClass()->getName(), $parameter->getDeclaringFunction()->getName(), $parameter->name); //@phpstan-ignore-line
-        $filePath = sprintf('/%s.php', str_replace('\\', '_', $index));
+        // ScriptName::from() rejects an unsafe index, so the interpolation below is safe
+        $filePath = sprintf('/%s.php', ScriptName::from($index));
         // Add prototype or singleton
         $this->args[] = $isSingleton ?
-            sprintf("\\Ray\\Compiler\\singleton(\$scriptDir, \$singletons, '%s', '%s', %s)", $index, $filePath, $ip) :
-            sprintf("\\Ray\\Compiler\\prototype(\$scriptDir, \$singletons, '%s', '%s', %s)", $index, $filePath, $ip);
+            sprintf("\Ray\Compiler\singleton(\$scriptDir, \$singletons, '%s', '%s', %s)", $index, $filePath, $ip) :
+            sprintf("\Ray\Compiler\prototype(\$scriptDir, \$singletons, '%s', '%s', %s)", $index, $filePath, $ip);
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/InstanceScript.php]: Good catch — confirmed and fixed in `059a0ec`.

You were right that validating the file name does not make the raw literal safe: a backslash cannot be rejected because class-string qualifiers (`annotatedWith(Foo::class)`) legitimately carry one. Emitted raw, a trailing backslash escaped the closing quote (ParseError in the compiled script) and consecutive backslashes collapsed, shifting the singleton key between compile time and runtime.

Reproduced both with failing tests first, then applied `var_export($index, true)` in `addDependencyArg()` and `pushAspectBind()` as suggested. The policy is now: validate what reaches the filesystem (`ScriptName`), escape what reaches generated code (`var_export`).
